### DISCUSSION TITLE
External redux state in state machine context synchronization

### DISCUSSION THREAD

**Discussion by acollazomayer** [statelyai/xstate#2801] [Q&A] [answered]
Hi, I am writing a wizard using the state as the stepper. Each step is a form and this is values are saved in redux-form. I need to show one step or the other depending on te value of the redux form. The idea is that each screen is dumb and only knows how to go next and how to go back.

The way that I've implemented this is by doing this:

`const [state, send] = useMachine(myMachine, {
  context: {
    form,
  },
  ...
})`

However this is a problem, because the guards are stuck with the value of initialization, the state machine context does not update when form changes.

Do you have any idea on how I can solve this? Thanks

**Comment by davidkpiano** [comment, marked as answer]
Yes, you can synchronize values by sending events to the machine when the values change:

```js
const [state, send] = useMachine(...);

useEffect(() => {
  send({ type: 'update.name', value: name });
}, [name]);
```

**Reply by acollazomayer** [reply]
Hi, @davidkpiano thanks for answering! We thought of that solution, however, I feel that it's a little bit hacky. If I may ask, why do you think this is the way to go?

**Reply by davidkpiano** [reply]
This is the way you would have to sync any two separate libraries that manage state; for example, if you wanted to sync react-query with Redux, you still have to `dispatch(...)` something to Redux when the data changes. Otherwise, the other library has no way of knowing that something changed.

This might be improved in the future with [the `input` feature](https://github.com/statelyai/xstate/pull/2779).
